### ISSUE TITLE
Core dependencies not mentioned on dependencies page

### ISSUE THREAD

**Issue by eguiraud** [root-project/web#379]
The problem: as a user I visited https://root.cern/install/dependencies, I have installed required as well as optional dependencies, and yet configuring a build with `cmake -Dfail-on-missing=ON ../root_src` does not work, because I'm missing dependencies (or I have to manually turn on their built-in versions.

The minimal working configuration I found was

```
cmake -Dbuiltin_zstd=ON -Dbuiltin_xxhash=ON -Dfail-on-missing=ON -Dbuiltin_lz4=ON -Dbuiltin_afterimage=ON -Dbuiltin_gl2ps=ON -Doracle=OFF -Dpythia6=OFF -Dpythia8=OFF -Dxrootd=OFF -Dgfal=OFF -Dbuiltin_tbb=ON -Dbuiltin_vdt=ON ../root_src
```

and note that if I set any of those builtins to OFF, configuration fails.

I think core dependencies that are required for ROOT to work, lest it cannot build, should be listed as dependencies in https://root.cern/install/dependencies.

From Oksana:

> in theory there should be libz liblzma liblz4 (it get xxhash as a dependency) and libzstd

I don't know where `gl2ps` or `vdt` lie.

**Comment by Axel-Naumann** [comment]
But IIUC and IMO these builtins are optional dependencies, as ROOT *can* build without them being installed, by taking its own.

**Comment by eguiraud** [comment]
@Axel-Naumann don't you agree that, after installing all dependencies listed at https://root.cern/install/dependencies, the following should work?

```
cmake -Dfail-on-missing=ON ../root_src
```

**Comment by couet** [comment]
@oshadura : I assigned it to you. Feel free to assign it to somebody else if you think that's not your domain.

**Comment by bellenot** [comment]
@eguiraud before I start to look at this, is it still relevant?

**Comment by Axel-Naumann** [comment]
Yes, the page does not list all required dependencies.

**Comment by Axel-Naumann** [comment]
@eguiraud 

> don't you agree that, after installing all dependencies listed at https://root.cern/install/dependencies, the following should work? `cmake -Dfail-on-missing=ON ../root_src`

Yes, after installing all *required* dependencies this command must work.

**Comment by bellenot** [comment]
OK, I'll check on all platforms
